src/main/app/common/country.ts

```diff
@@ -0,0 +1,122 @@
+
+export class PostcodeInfoCountry {
+  readonly GSS_CODE: string
+  readonly name: string
+}
+
+export class Country {
+  static readonly WALES = new Country('wales', 'Wales')
+  static readonly ENGLAND = new Country('england', 'England')
+  static readonly SCOTLAND = new Country('scotland', 'Scotland')
+  static readonly NORTHERN_IRELAND = new Country('northernIreland', 'Northern Ireland')
+
+  readonly value: string
+  readonly name: string
+
+  constructor (value: string, name: string) {
+    this.value = value
+    this.name = name
+  }
+
+  static all (): Country[] {
+    return [
+      Country.WALES,
+      Country.ENGLAND,
+      Country.SCOTLAND,
+      Country.NORTHERN_IRELAND
+    ]
+  }
+
+  static defendantCountries (): Country[] {
+    return [
+      Country.WALES,
+      Country.ENGLAND
+    ]
+  }
+
+  static async isValidClaimantAddress (form: Form<PartyDetails>): Promise<Form<PartyDetails>> {
+    await this.getCountry(form.model.address.postcode).then((country: string) => {
+      if (country.length > 0 && !Country.isClaimantCountry(country)) {
+        let error = new ValidationError()
+        error.property = 'address[postcode]'
+        error.constraints = { amount: 'The country must be England, Wales, Scotland or Northern Ireland' }
+        form.errors.push(new FormValidationError(error))
+      }
+    })
+
+    if (form.model.hasCorrespondenceAddress) {
+      await this.getCountry(form.model.address.postcode).then((country: string) => {
+        if (country.length > 0 && !Country.isClaimantCountry(country)) {
+          let error = new ValidationError()
+          error.property = 'correspondenceAddress[postcode]'
+          error.constraints = { amount: 'The country must be England, Wales, Scotland or Northern Ireland' }
+          form.errors.push(new FormValidationError(error))
+        }
+      })
+    }
+
+    return form
+  }
+
+  static async isValidDefendantAddress (form: Form<PartyDetails>): Promise<Form<PartyDetails>> {
+    await this.getCountry(form.model.address.postcode).then((country: string) => {
+      if (country.length > 0 && !Country.isDefendantCountry(country)) {
+        let error = new ValidationError()
+        error.property = 'address[postcode]'
+        error.constraints = { amount: 'The country must be England or Wales' }
+        form.errors.push(new FormValidationError(error))
+      }
+    })
+
+    if (form.model.hasCorrespondenceAddress) {
+      await this.getCountry(form.model.address.postcode).then((country: string) => {
+        if (country.length > 0 && !Country.isDefendantCountry(country)) {
+          let error = new ValidationError()
+          error.property = 'correspondenceAddress[postcode]'
+          error.constraints = { amount: 'The country must be England or Wales' }
+          form.errors.push(new FormValidationError(error))
+        }
+      })
+    }
+
+    return form
+  }
+
+  static isClaimantCountry (value: string): boolean {
```

Comment:
The method name is inaccurate as well. I would read the method name in the way `is provided country a claimant one` which is not what this method is doing.

Reply:
that is what this method is doing